rnnalyse/extractors/base_extractor.py

```diff
@@ -147,38 +145,76 @@ def _print_time_info(self, start_time: float, print_every: int) -> None:
               f'Time: {minutes:>3.0f}m {seconds:>2.2f}s\t'
               f'Speed: {speed:.2f}s/sen')
 
-    def _extract_sentence(self, sentence: Sentence) -> None:
+    def _extract_sentence(self, sentence: LabeledSentence) -> int:
         """ Generates the embeddings of a sentence and writes to file.
 
         Parameters
         ----------
         sentence : Sentence
             The to-be-extracted sentence, represented as a list of strings.
+
+        Returns
+        -------
+        num_extracted: int
+            Number of extracted activations for this sentence.
         """
+
+        sen_activations: PartialActivationDict = self._init_sen_activations(0)
```

Comment:
This might make perfect sense, but I cannot quickly figure out why previously this was the sentence length and now it is 0, because you don't know on forehand if all activations are required?

Reply:
Before, this was initialized as an empty np.array with dimensions sentence length x activation dim. Now that we have the selection function, we don't know in advance how many activations we will actually extract, that's why I opted to initialize it as 0 x activation dim instead and use `np.append`.